homeassistant/components/mqtt/config_flow.py

```diff
@@ -162,14 +159,12 @@ async def async_step_broker(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Confirm the setup."""
-        yaml_config: ConfigType = get_mqtt_data(self.hass, True).config or {}
```

Comment:
We do not need to get defaults from YAML since we already migrated the settings to the entry at `__init__.py`